### PR TITLE
Added Filter Functionality

### LINKED ISSUE #19: Add a filter functionality
What would be the reason for you not to take it over from the old project? If there is none, I'd gladly do it.

### PR DESCRIPTION
**Pull request by oldergod**
## Filter Functionality

Added a filter functionality so user can, as in the old css-triggers, filter entries.

### Demo

csstriggers__filter-demo

※ The design is based on https://www.google.com/design/spec/material-design/introduction.html

## Features

- Show/hide on the fly properties matching the filter, like in the old version.
- Handle `tabindex` for showen/hidden elements. i.e. works well with tab.
  - Aside with `aria-hidden` when needed.
- If only one entry is displayed when the form is submitted, it opens this entry, otherwise only `blur()`.

### Tools

- Use of material design icons.
- Use of FLIP for the animations.

### Room for improvement

- Manage scroll position while filtering as top matched property will be hidden behind the header. This can be seen at the end of the demo above. Might be improved/fixed if #18 get merged.
- Would be nice to open the filter on `/` keydown.

I tried to keep the code style as it was already.

fix #19

### CODE REVIEW

File: src/third_party/MaterialIcons/ic_close_white_18px.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg fill="#FFFFFF" height="18" viewBox="0 0 24 24" width="18" xmlns="http://www.w3.org/2000/svg">
+    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
+    <path d="M0 0h24v24H0z" fill="none"/>
+</svg>
```
Comment (surma) [inline src/third_party/MaterialIcons/ic_close_white_18px.svg]: These 2 files need to be added to `sw.js` for offline support

**Comment by surma** [comment]
@oldergod That is some A+ work right here! Awesome.

Just one minor thing and then this should be good to go :)

**Comment by oldergod** [comment]
@surma I totally forgot about him !

It is fixed. Thank you.

File: src/styles/core.scss [hunk 2]
```diff
@@ -643,6 +643,7 @@ body {
   position: absolute;
   border-radius: 2px;
   padding: 2px 0;
+  pointer-events: auto;
```
Comment (oldergod) [inline src/styles/core.scss]: because header not expanded is `pointer-events: none;`

**Pull request merged**